c++/src/MemoryPool.cc

```diff
@@ -88,7 +88,7 @@ namespace orc {
 
   template <class T>
   void DataBuffer<T>::reserve(uint64_t newCapacity){
-    if (newCapacity > currentCapacity) {
+    if (newCapacity > currentCapacity || !buf) {
```

Comment:
This is a redundant condition that will never be used. Does Coverity complain about this?